### ISSUE TITLE
Trying to create a namespace even though it exists

### ISSUE THREAD

**Issue by jin-harmoney** [garden-io/garden#3696]
### Current Behavior

Error when running `garden deploy`:

```
Failed getting status for service 'web' (from module 'web'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Namespace harmoney-remote-hno doesn't exist and Garden was unable to create it. You may need to create it manually or ask an administrator to do so.
```

This is not the correct error though, because the namespace exists. From the silly logs:

```
...
[silly] Getting getServiceStatus handler for service api-redirect-hook
[silly] Getting getServiceStatus handler for service cdn
[silly] Getting getServiceStatus handler for service db
[silly] Getting getServiceStatus handler for service pdf
[silly] Getting getServiceStatus handler for service web
[silly] Getting getServiceStatus handler for service api
[silly] Calling getServiceStatus handler for service api-redirect-hook
[silly] Calling getServiceStatus handler for service cdn
[silly] Calling getServiceStatus handler for service db
[silly] Calling getServiceStatus handler for service pdf
[silly] Calling getServiceStatus handler for service web
[silly] Calling getServiceStatus handler for service api
...
[verbose] Creating namespace harmoney-remote-hno
...
[silly] Getting getServiceStatus handler for service listcheck
[silly] Calling getServiceStatus handler for service listcheck
...
Failed getting status for service 'web' (from module 'web'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Namespace harmoney-remote-hno doesn't exist and Garden was unable to create it. You may need to create it manually or ask an administrator to do so.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
[silly] Error: Namespace harmoney-remote-hno doesn't exist and Garden was unable to create it. You may need to create it manually or ask an administrator to do so.
    at ensureNamespace (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\plugins\kubernetes\namespace.ts:89:15)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at getNamespaceStatus (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\plugins\kubernetes\namespace.ts:169:5)
    at Object.getServiceStatus (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\plugins\kubernetes\helm\status.ts:59:27)
    at C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\actions.ts:1303:24
    at ActionRouter.callServiceHandler (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\actions.ts:1123:40)
    at ActionRouter.getServiceStatus (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\actions.ts:529:24)
    at wrapped.process (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\tasks\get-service-status.ts:129:16)
    at TaskNode.process (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\task-graph.ts:801:20)
    at wrapped.processNode (C:\snapshot\project\tmp\pkg\cli\node_modules\@garden-io\core\src\task-graph.ts:436:18)

Error Details:

error:
  detail:
    body:
      kind: Status
      apiVersion: v1
      metadata: {}
      status: Failure
      message: namespaces "harmoney-remote-hno" already exists
      reason: AlreadyExists
      details:
        name: harmoney-remote-hno
        kind: namespaces
      code: 409
    request:
      method: POST
      useQuerystring: false
      body: >-
        {"apiVersion":"v1","kind":"Namespace","metadata":{"annotations":{"garden.io/generated":"true","garden.io/version":"0.12.48"},"name":"harmoney-remote-hno"}}
      readable: true
      writable: true
      explicitMethod: true
      proxy: null
      tunnel: true
      setHost: true
      port: 443
      host: 35.189.218.181
      path: /api/v1/namespaces
      href: 'https://35.189.218.181/api/v1/namespaces'
      ntick: true
      originalHost: 35.189.218.181
      originalHostHeaderName: host
  type: kubernetes
  statusCode: 409
```

So you can see that the k8s api is returning a `409` with `message: namespaces "harmoney-remote-hno" already exists`.

### Expected behavior

The namespace is not created because it already exists -> there is no error.

### Reproducible example

I'm afraid this doesn't happen consistently, so there is no way to reproduce this. A retry of `garden deploy` typically works.

### Your environment

* **OS:** Happens on Windows, Linux and MacOS
* **How I'm running Kubernetes:** GKE

garden version: 0.12.48

I searched for related issues, but didn't find any.

**Comment by vvagaytsev** [comment]
@jin-harmoney thanks for reporting this! We'll take a look.

**Comment by vvagaytsev** [comment]
We might need to fix it by ignoring the error if the target namespace already exists.

**Comment by stefreak** [comment]
This happened to me too now, here's how to reproduce it:

deploy any demo project (garden deploy)
Then, delete the namespace (e.g. in Lens) and at the same time, run garden deploy again. 

The namespace will be in the state "terminating", garden will try to create a new one with the same name and fail.

**Comment by vvagaytsev** [comment]
Closing as fixed in #4248. Feel free to reopen if it's still an issue.
